src/realm/sync/noinst/client_impl_base.hpp

```diff
@@ -1262,8 +1266,10 @@ inline void ClientImpl::Connection::one_more_active_unsuspended_session()
 
 inline void ClientImpl::Connection::one_less_active_unsuspended_session()
 {
+    REALM_ASSERT(m_num_active_unsuspended_sessions);
```

Comment:
I've added these assertions wherever we -- a variable because there was at least one case where we were underflowing and making this value uint64_t(-1).

Reply:
I like having the assertions there - it will help with tracking down any issues in the future.